### ISSUE TITLE
Values merge where a key with an nil value is added does not work

### ISSUE THREAD

**Issue by mattymo** [roboll/helmfile#1154]
Problem: Adding a new key to a map with a nil value doesn't merge.

Scenario to reproduce

values1.yaml:
```
components:
  etcd-operator:
   version: 0.10.0
```

values2.yaml:
```
components:
  etcd-operator:
    version:  0.10.3
  prometheus:
```

helmfile.yaml
```
repositories:
- name: "stable"
  url: "https://kubernetes-charts.storage.googleapis.com"

environments:
  default:
    values:
      - values1.yaml
      - values2.yaml

releases:
- name: "etcd-operator"
  chart: stable/etcd-operator
{{- if .Values.components | getOrNil "etcd-operator" | getOrNil "version" }}
  version: {{ .Values.components  | getOrNil "etcd-operator"  | getOrNil "version"  }}
{{- end }}

{{- if hasKey .Values.components "prometheus }}
- name: "prometheus"
  chart: stable/prometheus
{{- if .Values.components | getOrNil "prometheus" | getOrNil "version" }}
  version: {{ .Values.components  | getOrNil "prometheus"  | getOrNil "version"  }}
{{- end }}
{{- end }}
```

Expected calculated values:
```
components:
  etcd-operator:
    version:  0.10.3
  prometheus:
```

Actual calculated values:
```
components:
  etcd-operator:
    version:  0.10.3
```

Note: If prometheus is an empty map {}, it works, or if prometheus has some version set, it also works.

Helmfile version: v0.104.0

**Comment by mumoshu** [comment]
@mattymo Thanks! Seems like this is resolved in mergo via https://github.com/imdario/mergo/pull/133. I'll bring it to helmfile soon.

**Comment by mumoshu** [comment]
@mattymo This should be fixed in v0.106.2. Could you confirm if it works on your machine, too?
